server/src/user/model.rs

```diff
@@ -187,17 +187,24 @@ impl User {
     pub async fn set_email<'a, E>(
         payload: &user_set_email_mutation::Payload,
         executor: E,
-    ) -> Result<(), sqlx::Error>
+    ) -> Result<String, sqlx::Error>
     where
         E: Executor<'a>,
     {
+        let mut transaction = executor.begin().await?;
+
+        let username = sqlx::query!("select username from user where id = ?", payload.user_id)
```

Comment:
That does the job, that shouldn't be anyway complicated. My tendency would be to use a method of the uuid::model::user model.

Reply:
> That does the job, that shouldn't be anyway complicated. My tendency would be to use a fetch method of the uuid::model::user model. Do you think it is easy to use a fetch method instead?

can you try to implement it?

Reply:
> can you try to implement it?
I'll try in another moment, putting in my TODO list.

Reply:
I've managed to do it locally, but it's not worthy, too much code for too little. Let's leave as it is.